asmcomp/compilenv.mli

```diff
@@ -14,7 +14,7 @@
 
 open Cmx_format
 
-val reset: ?packname:string -> string -> unit
+val reset: ?packname:string -> sourcefile:Timings.build_kind -> string -> unit
```

Comment:
`build_kind` is probably not the best name for that type we can come up with.
If you are better than me at naming, it is a value describing the the origin compilation.
In most case it is `File somesome_filename`. But linking also generates code that goes through the backend. In those cases it can be either `Startup` when linking a program or `Pack pack_name` when linking a package.

I chose to name the label `sourcefile` as it has the same purpose as the other arguments labeled the same way: propagating information about the origin of the compilation for debugging purpose. It may be that `sourcefile` is not the right way to name it.